File: doc-examples/mongo-example-java/src/main/java/example/AbstractBookRepository.java

package example;

import io.micronaut.data.mongodb.annotation.MongoRepository;
import io.micronaut.data.repository.CrudRepository;
import org.bson.types.ObjectId;

import java.util.List;

@MongoRepository
public abstract class AbstractBookRepository implements CrudRepository<Book, ObjectId> {

    public abstract List<Book> findByTitle(String title);
}

File: doc-examples/mongo-example-python/src/main/python/example/AbstractBookRepository.py
from micronaut.data.mongodb.annotation import MongoRepository
from micronaut.data.repository import CrudRepository
from org.bson.types import ObjectId

from example.Book import Book


@MongoRepository
class AbstractBookRepository(CrudRepository[Book, ObjectId]):

    def findByTitle(self, title: str) -> list[Book]: ...
